src/coffee-script.coffee

```diff
@@ -63,6 +63,13 @@ exports.compile = compile = withPrettyErrors (code, options) ->
     token[1] for token in tokens when token[0] is 'IDENTIFIER'
   )
 
+  # Check for import or export; if found, force bare mode
+  unless options.bare? and options.bare is yes
+    for token in tokens
+      if token[0] in ['IMPORT', 'EXPORT']
+        options.bare = yes
```

Comment:
We haven’t written the documentation for this feature yet. When we do, we will note that using modules forces `bare` mode automatically for the files in which modules are imported. As @lydell points out, users might not want `bare` mode for all files in a project, and the flag doesn’t provide a way to enable it selectively. Simply telling people to set `bare` to true isn’t an option for that reason.

We will also be writing big warnings about how CoffeeScript won’t be resolving modules and therefore Babel or similar will be necessary. This will all get covered. Thanks for pointing it out, though, it’s important that we don’t forget to note this.